Source/Core/VideoBackends/D3D/D3DBoundingBox.cpp

```diff
@@ -14,45 +14,39 @@
 
 namespace DX11
 {
-static constexpr u32 NUM_BBOX_VALUES = 4;
-static ComPtr<ID3D11Buffer> s_bbox_buffer;
-static ComPtr<ID3D11Buffer> s_bbox_staging_buffer;
-static ComPtr<ID3D11UnorderedAccessView> s_bbox_uav;
-static std::array<s32, NUM_BBOX_VALUES> s_bbox_values;
-static std::array<bool, NUM_BBOX_VALUES> s_bbox_dirty;
-static bool s_bbox_valid = false;
-
-ID3D11UnorderedAccessView* BBox::GetUAV()
+D3DBoundingBox::~D3DBoundingBox()
 {
-  return s_bbox_uav.Get();
+  m_uav.Reset();
+  m_staging_buffer.Reset();
+  m_buffer.Reset();
 }
 
-void BBox::Init()
+bool D3DBoundingBox::Initialize()
 {
   if (!g_ActiveConfig.backend_info.bSupportsBBox)
```

Comment:
This check happens in every backend (save for D3D12, Software, and Null).  Should it be moved to the abstract interface?  Vulkan supports a log message which might be nice in general.

Additionally, why is this returning `true`?  If it's not supported, seems like we would want to error out or possibly return a `NullBoundingBox` to avoid errors when it is being used.

Reply:
Read/Write also check this same config option (but not in all backends).

Reply:
Returning `false` would cause us to abort backend initialization, which we don't want because games can still run without bounding box support.

Reply:
Ah ok.  Well, I'd let me amend my suggestion then to modifying Read in `AbstractBoundingBox` to:

```
std::vector<BBoxType> Read(int index, int length)
{
  if (!g_ActiveConfig.backend_info.bSupportsBBox)
    return {};

  return ReadImpl(index, length);
}
```

with a

```
private:
virtual std::vector<BBoxType> ReadImpl(int index, int length) = 0;
```

and same with `Write()` so that all backends get this behavior in a single place.

And I'd add that check to D3D12's `Initialize()` for consistency with D3D11, OpenGL, and Vulkan.

Reply:
I already addressed it in a slightly different way, but thanks. Didn't want to create forwarding functions, as the implementation here is pretty tight anyways.

D3D12 implicitly supports bounding box, so the check would be kinda pointless there. In any case I moved the initialization check to RenderBase.